x-pack/plugins/security_solution/public/timelines/components/row_renderer_switch/index.tsx

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { EuiSwitchEvent } from '@elastic/eui';
+import { EuiToolTip, EuiSwitch, EuiFormRow, useGeneratedHtmlId } from '@elastic/eui';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import styled from 'styled-components';
+import { RowRendererId } from '../../../../common/api/timeline';
+import type { State } from '../../../common/store';
+import { setExcludedRowRendererIds } from '../../store/actions';
+import { selectExcludedRowRendererIds } from '../../store/selectors';
+import * as i18n from './translations';
+
+interface RowRendererSwitchProps {
+  timelineId: string;
+}
+
+const CustomFormRow = styled(EuiFormRow)`
+  .euiFormRow__label {
+    font-weight: 400;
+  }
+`;
+
+export const RowRendererSwitch = React.memo(function RowRendererSwitch(
+  props: RowRendererSwitchProps
+) {
+  const toggleTextSwitchId = useGeneratedHtmlId({ prefix: 'rowRendererSwitch' });
+
+  const { timelineId } = props;
+
+  const dispatch = useDispatch();
+
+  const excludedRowRendererIds = useSelector((state: State) =>
+    selectExcludedRowRendererIds(state, timelineId)
+  );
+
+  const areAllRowRenderersExcluded = useMemo(
```

Comment:
```suggestion
  const areAllRowRenderersExcluded = useMemo(() => Object.values(RowRendererId).length === excludedRowRendererIds.length),
    [excludedRowRendererIds]
  );
```

or if you do want to iterate you can do
```suggestion
const areAnyRowRenderersEnabled = useMemo(() => Object.values(RowRendererId).some(id => !excludedRowRendererIds.includes(id))
```
and get the benefit of not iterating over the whole array if the first one is enabled or something, then you don't have to check for a negation on line 47

Reply:
Thanks. I like the second option as it gives the surity. Done here: 1c07c657c07